src/pages/_app.tsx

```diff
@@ -105,32 +103,30 @@ export default function App({
 		<QueryClientProvider client={queryClient}>
 			<SSRProvider>
 				<CurrentContentTypeProvider currentContentType={currentContentType}>
-					<ErrorBoundary FallbackComponent={DevDotClient}>
```

Comment:
The clientside 404 link error was always being caught by this. But the `DevDotClient` doesn't render a layout at all, so the deadend is not preferred UX. 

I'd love feedback here on the reason why we need this in addition to the custom _error page. It seems like next will automatically router everything to the custom error route in the cases we would need for this error boundary. But would love to learn more there. 

The `error-view-switcher` component has the logic to know how to handle the error view properly, so really we want to route folks there in this case.

Reply:
> ErrorBoundary is ideal for handling client-side errors. Without ErrorBoundary, it's hard (impossible?) to catch and log the details of a client-side error.

Yep. The distinction here is errors that are thrown while a component is rendering (as you're seeing in your test PR!).

> pages/404.js & pages/500.js are a way to create custom pages to handle those specific server-side errors
Yep!

> pages/_error.js can handle both client-side and server-side errors... but we won't have the error details we'd otherwise get with ErrorBoundary?

It depends on when the error happens. If we get an error back when fetching page data on navigation from the client, then yeah I think Next would route to our custom error page, but any errors that are thrown during render would not get handled without the error boundary. `_error` is synonymous with the `404.js` / `500.js` error pages, it just allows for SSRing the error page instead of relying on a statically generated one.

Reply:
Interesting, so maybe we could make the error boundary for clientside errors 'smarter' in the same way the _error page is to render the right view.

Reply:
Maybe we could create a unique error boundary for the different views — .io versus dev portal.